packages/@react-aria/slider/src/useSlider.ts

```diff
@@ -67,7 +67,9 @@ export function useSlider<T extends number | number[]>(
   const realTimeTrackDraggingIndex = useRef<number | null>(null);
 
   const stateRef = useRef<SliderState>(null);
-  stateRef.current = state;
+  useEffect(() => {
+    stateRef.current = state;
+  });
```

Comment:
There must have been a reason we wrote it in render before... any idea? I thought usually it was because effects didn't update fast enough (e.g. before events) sometimes so things would get out of sync. that's not a slider specific thing just in general - I remember things like that in useMove/usePress too. so we should be careful I guess

Reply:
yeah, i'm not sure we truly did it on purpose so much so as it is hard to think through the order of events when you can't update things in render

if we never read from stateRef during render though, this should be plenty fast enough
and we don't pass this stateRef off to anything else

I think the useEffect is always run before the next render cycle, so it should be at the correct value

Reply:
useEffect runs asynchronously though, so it might be possible for an event to fire between when a component renders and when the ref updates. Haven't tested that though.

- render
- event
- useEffect

Maybe we should use useLayoutEffect?